src/components/Sweep/SweepExecute.tsx

```diff
@@ -0,0 +1,202 @@
+import { useCallback, useState, useEffect } from 'react';
+import { Form, Input, Button, notification, Typography } from 'antd';
+import { ethers } from 'ethers';
+import styled from 'styled-components';
+
+import { ethereumAddressValidator } from '../../utils';
+import { useMetamask } from '../../hooks/useMetamask';
+import { ChainId } from '../../constants';
+import { contractBalanceOf, contractTransferFrom, contractAllowance } from './useSweep';
+
+const { Text } = Typography;
+
+function SweepExecute() {
+  const [loading, setLoading] = useState(false);
+  const [valid, setValid] = useState(false);
+
+  const [fromAddress, setFromAddress] = useState('');
+  const [to, setTo] = useState('');
+  const [amount, setAmount] = useState(0);
+  const [balance, setBalance] = useState(0);
+  const [allowance, setAllowance] = useState(0);
+
+  const [form] = Form.useForm();
+
+  const { changeNetwork } = useMetamask();
+
+  const handleFromChange = useCallback(
+    (e) => {
+      setFromAddress(e.target.value);
+    },
+    [setFromAddress]
+  );
+
+  useEffect(() => {
+    const validAFromAddress = ethers.utils.isAddress(fromAddress);
+    const getBalance = async () => {
+      const balanceEth = await contractBalanceOf(fromAddress);
+      setBalance(balanceEth[1]);
+    };
+
+    const validToAddress = ethers.utils.isAddress(to);
+    const getAllowance = async () => {
+      const balanceWei = await contractAllowance(fromAddress, to);
+      setAllowance(balanceWei);
+    };
+
+    if (validAFromAddress) {
+      try {
+        getBalance();
```

Comment:
you have an async function in `try ... catch` but you're not `await`ing it so no errors will be caught.
also, if you want to use `try ... catch` then move it into `getBalance` and `getAllowance`. this will make the code more readable